src/Dispatch/Dispatcher.php

```diff
@@ -124,7 +124,7 @@ public function removeListener(string $event, HandlerInterface $handler): self
     public function removeListenerByClassName(string $event, string $className): self
     {
         $handlers = $this->handlers[$event] ?? [];
-        $this->handlers = array_filter($handlers, function ($existing) use ($className) {
+        $this->handlers[$event] = array_filter($handlers, function ($existing) use ($className) {
```

Comment:
This fixes the issue where custom loaders would completely break the handlers configuration.